datasets/visual_genome/visual_genome.py

```diff
@@ -0,0 +1,465 @@
+    """Normalizes attributes annotation in-place"""
+    # Some attributes annotations don't have an attribute field
+    for attribute in annotation["attributes"]:
+        # `id` should be converted to `object_id`:
+        if "id" in attribute:
+            attribute["object_id"] = attribute["id"]
+            del attribute["id"]
+
+        # `objects_names` should be convered to `names:
+        if "object_names" in attribute:
+            attribute["names"] = attribute["object_names"]
+            del attribute["object_names"]
+
+        # Some versions of `attribute` annotations don't have `synsets` field.
+        if "synsets" not in attribute:
+            attribute["synsets"] = None
+
+        # Some versions of `attribute` annotations don't have `attributes` field.
+        if "attributes" not in attribute:
+            attribute["attributes"] = None
+
+    return annotation
+
+
+def _normalize_relationship_annotation_(annotation: Dict[str, Any]) -> Dict[str, Any]:
+    """Normalizes relationship annotation in-place"""
+    # For some reason relationships objects have a single name instead of a list of names.
+    for relationship in annotation["relationships"]:
+        # `id` should be converted to `object_id`:
+        if "id" in relationship:
+            relationship["relationship_id"] = relationship["id"]
+            del relationship["id"]
+
+        if "synsets" not in relationship:
+            relationship["synsets"] = None
+
+        subject = relationship["subject"]
+        object_ = relationship["object"]
+
+        for obj in [subject, object_]:
+            # `id` should be converted to `object_id`:
+            if "id" in obj:
+                obj["object_id"] = obj["id"]
+                del obj["id"]
+
+            if "name" in obj:
+                obj["names"] = [obj["name"]]
+                del obj["name"]
+
+            if "synsets" not in obj:
+                obj["synsets"] = None
+
+    return annotation
+
+
+def _normalize_image_metadata_(image_metadata: Dict[str, Any]) -> Dict[str, Any]:
+    """Normalizes image metadata in-place"""
+    if "id" in image_metadata:
+        image_metadata["image_id"] = image_metadata["id"]
+        del image_metadata["id"]
+    return image_metadata
+
+
+_ANNOTATION_NORMALIZER = defaultdict(lambda: lambda x: x)
+_ANNOTATION_NORMALIZER.update(
+    {
+        "region_descriptions": _normalize_region_description_annotation_,
+        "objects": _normalize_object_annotation_,
+        "attributes": _normalize_attribute_annotation_,
+        "relationships": _normalize_relationship_annotation_,
+    }
+)
+
+# ---- Visual Genome loading script ----
+
+
+class VisualGenomeConfig(datasets.BuilderConfig):
+    """BuilderConfig for Visual Genome."""
+
+    def __init__(self, name: str, version: Optional[str] = None, with_image: bool = True, **kwargs):
+        _version = _LATEST_VERSIONS[name] if version is None else version
+        _name = f"{name}_v{_version}"
```

Comment:
After much thought into this, I don't think this makes sense anymore. If preview/autotrain rely on only names it's probably the issue of those components no? I think we should have a single name that has a default version (1.2.0) in our case.

Reply:
What do you mean ? I think it's fine this way ?

Reply:
I really don't like the use of versions in the naming, and I'd rather have it as a config argument (I usually don't care about the version, I usually only care about about the task ...)